### PR TITLE
515 enable read/write of bool arrays

### PR DESCRIPTION
**Pull request by hokiegeek2**

### CODE REVIEW

File: tests/io_test.py [hunk 1]
```diff
@@ -83,14 +89,16 @@ def testSaveAllLoadAllWithDict(self):
...
                          retrieved_columns['int_tens_pdarray'].all())
         self.assertEqual(self.dict_columns['int_hundreds_pdarray'].all(), 
                          retrieved_columns['int_hundreds_pdarray'].all())
         self.assertEqual(self.dict_columns['float_pdarray'].all(), 
-                         retrieved_columns['float_pdarray'].all())      
-        self.assertEqual(3, len(ak.get_datasets('{}/iotest_dict_LOCALE0'.format(IOTest.io_test_dir))))
+                         retrieved_columns['float_pdarray'].all())    
+        self.assertEqual(self.dict_columns['bool_pdarray'].all(), 
+                         retrieved_columns['bool_pdarray'].all())    
+        self.assertEqual(4, len(ak.get_datasets('{}/iotest_dict_LOCALE0'.format(IOTest.io_test_dir))))
```
Comment (reuster986) [inline tests/io_test.py:101]: These `assertEqual` tests should use the form:
```python
self.assertEqual((self.dict_columns['name'] == retrieved_columns['name']).all())
```
In their current form, the tests will not actually compare the values of the arrays.

File: tests/io_test.py [hunk 2]
```diff
@@ -105,14 +113,16 @@ def testSaveAllLoadAllWithList(self):
...
-        self.assertEqual(3, len(retrieved_columns))
+        self.assertEqual(4, len(retrieved_columns))
         self.assertEqual(self.list_columns[0].all(), 
                          retrieved_columns['int_tens_pdarray'].all())
         self.assertEqual(self.list_columns[1].all(), 
                          retrieved_columns['int_hundreds_pdarray'].all())
         self.assertEqual(self.list_columns[2].all(), 
                          retrieved_columns['float_pdarray'].all())      
-        self.assertEqual(3, len(ak.get_datasets('{}/iotest_list_LOCALE0'.format(IOTest.io_test_dir))))
+        self.assertEqual(self.list_columns[3].all(),
```
Comment (reuster986) [inline tests/io_test.py:123]: same issue with this block of tests

File: tests/io_test.py [hunk 3]
```diff
@@ -199,10 +209,11 @@ def testReadAllWithGlob(self):
         
         dataset = ak.read_all(filenames='{}/iotest_dict_columns*'.format(IOTest.io_test_dir))
 
-        self.assertEqual(3, len(list(dataset.keys())))  
+        self.assertEqual(4, len(list(dataset.keys())))  
         self.assertEqual(self.int_tens_pdarray.all(), dataset['int_tens_pdarray'].all())
         self.assertEqual(self.int_hundreds_pdarray.all(), dataset['int_hundreds_pdarray'].all())
         self.assertEqual(self.float_pdarray.all(), dataset['float_pdarray'].all())
+        self.assertEqual(self.bool_pdarray.all(), dataset['bool_pdarray'].all())
```
Comment (reuster986) [inline tests/io_test.py:216]: same issue here

File: tests/io_test.py [hunk 4]
```diff
@@ -221,10 +232,13 @@ def testLoad(self):
                                         dataset='int_hundreds_pdarray')
         result_array_float = ak.load(path_prefix='{}/iotest_dict_columns'.format(IOTest.io_test_dir), 
                                      dataset='float_pdarray')
+        result_array_bool = ak.load(path_prefix='{}/iotest_dict_columns'.format(IOTest.io_test_dir), 
+                                     dataset='bool_pdarray')
 
         self.assertEqual(self.int_tens_pdarray.all(), result_array_tens.all())
         self.assertEqual(self.int_hundreds_pdarray.all(), result_array_hundreds.all())
         self.assertEqual(self.float_pdarray.all(), result_array_float.all())
+        self.assertEqual(self.bool_pdarray.all(), result_array_bool.all())
```
Comment (reuster986) [inline tests/io_test.py:241]: same issue

File: src/GenSymIO.chpl [hunk 5]
```diff
@@ -466,7 +466,22 @@ module GenSymIO {
...
+                    /*
+                     * Since boolean pdarrays are saved to and read from HDF5 as ints, confirm whether this
+                     * is actually a boolean dataset. If so, (1) convert the SymEntry pdarray to a boolean 
+                     * pdarray, (2) create a new SymEntry of type bool, (3) set the SymEntry pdarray 
+                     * reference to the bool pdarray, and (4) add the entry to the SymTable
+                     */
+                    if isBooleanDataset(filenames[0],dsetName) {
+                        var a_bool = entryInt.a:bool;
+                        var entryBool = new shared SymEntry(len, bool);
+                        entryBool.a = a_bool;
```
Comment (reuster986) [inline src/GenSymIO.chpl:479]: You can get rid of a copy here by deleting line 477 and changing 479 to:

```chpl
entryBool.a = entryInt.a:bool;
```

**Review by reuster986 — CHANGES_REQUESTED** [review]
A few issues with the tests, but I think the logic looks good.

File: src/GenSymIO.chpl (hunk 5 above)
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply src/GenSymIO.chpl]: Good point, fixed and pushed to git

Comment (hokiegeek2) [reply src/GenSymIO.chpl]: Good point, fixed and pushed to git

File: tests/io_test.py (hunk 4 above)
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py (hunk 3 above)
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py (hunk 2 above)
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py (hunk 1 above)
Comment (hokiegeek2) [reply tests/io_test.py:101]: fixed and pushed to git

**Review by mhmerrill — COMMENTED** [review]
things look good to me but I'll defer to Bill for approval and merging

**Review by reuster986 — APPROVED** [review]
looks good!

**Comment by reuster986** [comment]
I'll let the checks finish before merging.

**Pull request merged**